### ISSUE TITLE
S2259 does not always catch nulls incorrectly guarded by `typof` checks

### ISSUE THREAD

**Issue by rkrisztian** [SonarSource/SonarJS#1737]
**I want to report a bug.**

**SonarJS version:** 5.2.1 (build 7778)
**SonarQube version:** Enterprise Edition 7.7 (build 23042)
**SonarLint version:** N/A

**Rule key:** javascript:S2259

**Reproducer** 
```js
let myObj = null;
if (typeof myObj === 'object') {
	console.log(myObj.myVar); // False positive here
}
```

This code, when executed in the browser, throws:

```text
TypeError: myObj is null
```

Reasoning: https://stackoverflow.com/questions/20480729/why-does-typeofnull-return-object-but-you-cant-assign-properties-to-it

**Expected behavior**

SonarJS should raise an error like:

> TypeError can be thrown as "myObj" might be null or undefined here.

Sometimes the detection works, for example in this "less trivial" case:

```js
function getSomeValue(someObj, someProp) {
	let someValue = someObj && someObj[someProp];
	if (typeof someValue === 'object') {
		someValue = someValue.someInnerProp; // True positive here
	}

	return someValue ? someValue : 'no value';
}
```

(`getSomeValue({ myProp: null }, 'myProp')` would throw the error.)

I can't really imagine why the rule works differently in the two cases.

**Comment by vilchik-elena** [comment]
We changed this rule implementation, and this FP is not relevant anymore
